streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignorTest.java

```diff
@@ -166,12 +166,18 @@ public void shouldAssignTasksEvenlyWithUnequalTopicGroupSizes() {
 
         assertThat(assign(TASK_1_0, TASK_0_0, TASK_0_1, TASK_0_2, TASK_0_3, TASK_0_4, TASK_0_5), is(false));
 
-        final Set<TaskId> expectedClientITasks = new HashSet<>(asList(TASK_0_0, TASK_0_1, TASK_1_0, TASK_0_5));
-        final Set<TaskId> expectedClientIITasks = new HashSet<>(asList(TASK_0_2, TASK_0_3, TASK_0_4));
+        final Set<TaskId> allTasks = new HashSet<>(asList(TASK_0_0, TASK_0_1, TASK_1_0, TASK_0_5, TASK_0_2, TASK_0_3, TASK_0_4));
```

Comment:
It might seem weird that some of these tests started failing since this PR is just adding an unused API, but it's just because they're very rigid and tend to assert a specific task assignment rather than validating the assignment contract, eg that tasks are evenly balanced. I'm guessing that just adding a field to the TaskId class, even though it's unused, resulted in a different hash and therefore ordering in some unsorted data structure somewhere in the assignor. 

I did verify that the new test results are still correct within the expected constraints, then adapted the tests to be more durable rather than just modifying the expected task assignments

Reply:
That makes sense, however you might be able not include the new field from the hash to prevent a chaotic assignment if you wanted